equinix/resource_network_device.go

```diff
@@ -1508,6 +1553,28 @@ func uploadDeviceLicenseFile(openFunc openFile, uploadFunc uploadLicenseFile, ty
 	return nil
 }
 
+func uploadDeviceCloudInitFile(openFunc openFile, uploadFunc uploadFile, typeCode string, device *ne.Device) error {
+	if device == nil || ne.StringValue(device.CloudInitFile) == "" {
```

Comment:
Taking the cloud-init contents as a parameter, instead of the file name, would enable us to remove this file-loading code.